crates/wasmprinter/src/operator.rs

```diff
@@ -1286,3 +1341,450 @@ impl<'a> VisitOperator<'a> for PrintOperator<'_, '_, '_, '_> {
+	    let mut buf_color = PrintTermcolor(Ansi::new(Vec::new()));
+	    let mut buf_nocolor = PrintTermcolor(NoColor::new(Vec::new()));
+	    let internal_config = Config { print_offsets: false,
+					   print_skeleton: false,
+					   name_unnamed: self.printer.config.name_unnamed,
+					   fold_instructions: false };
+	    let mut internal_printer = Printer { config: &internal_config,
+						 result: if use_color { &mut buf_color } else { &mut buf_nocolor },
+						 nesting: self.printer.nesting,
+						 line: self.printer.line,
+						 group_lines: Vec::new(),
+						 code_section_hints: Vec::new() };
+
+	    let mut op_printer = PrintOperator::new(&mut internal_printer, self.state, self.operator_state);
+	    define_visit!(before_op op_printer $op);
+
+	    op_printer.push_str(define_visit!(name $op))?;
+            $(
+                define_visit!(payload op_printer $op $($arg)*);
+            )?
+            define_visit!(after_op op_printer $op);
+
+	    self.printer.nesting = internal_printer.nesting;
+            self.printer.line = internal_printer.line;
+
+	    let _inst = String::from_utf8(if use_color { buf_color.0.into_inner() } else { buf_nocolor.0.into_inner() }).expect("invalid UTF-8");
+	    define_visit_folded!(handle $op self blockty _inst _params _results)
+        }
+    )*);
+
+    (get_block_type Loop      $var:ident $ty:ident) => {let $var = $ty;};
+    (get_block_type Block     $var:ident $ty:ident) => {let $var = $ty;};
+    (get_block_type If        $var:ident $ty:ident) => {let $var = $ty;};
+    (get_block_type TryTable  $var:ident $ty:ident) => {let $var = $ty.ty;};
+    (get_block_type $op:ident $var:ident $($_:tt)*) => {};
+
+    (handle Loop      $self:ident $ty:ident $inst:ident $params:ident $results:ident) => {$self.push_block($ty, FrameKind::Loop, $inst)};
+    (handle Block     $self:ident $ty:ident $inst:ident $params:ident $results:ident) => {$self.push_block($ty, FrameKind::Block, $inst)};
+    (handle TryTable  $self:ident $ty:ident $inst:ident $params:ident $results:ident) => {$self.push_block($ty, FrameKind::TryTable, $inst)};
+    (handle If        $self:ident $ty:ident $inst:ident $params:ident $results:ident) => {$self.push_if($ty, $inst)};
+
+    (handle Else      $self:ident $ty:ident $inst:ident $params:ident $results:ident) => {$self.handle_else()};
+    (handle End       $self:ident $ty:ident $inst:ident $params:ident $results:ident) => {$self.handle_end($results)};
+
+    (handle Try       $self:ident $ty:ident $inst:ident $params:ident $results:ident) => {bail!("legacy-exceptions not supported")};
+    (handle Catch     $self:ident $ty:ident $inst:ident $params:ident $results:ident) => {bail!("legacy-exceptions not supported")};
+    (handle CatchAll  $self:ident $ty:ident $inst:ident $params:ident $results:ident) => {bail!("legacy-exceptions not supported")};
+    (handle Delegate  $self:ident $ty:ident $inst:ident $params:ident $results:ident) => {bail!("legacy-exceptions not supported")};
```

Comment:
If you'd like to implement support for this you're more than welcome to do so, but if you'd prefer to leave this out that's also totally fine. Regardless I agree that as-is is ok where an error is returned.

Reply:
wasm-tools doesn't parse the folded text format for legacy-exceptions, so it seemed like a pain to implement it and I wasn't sure I'd get it right without tests. :-)